src/cliargs.lua

```diff
@@ -1,666 +1,41 @@
-local cli, _
+local cli = require('cliargs.core')()
+local deprecated = require('cliargs.utils.deprecated')
 
--- ------- --
--- Helpers --
--- ------- --
+-- Clean up the entire module (unload the scripts) as it's expected to be
+-- discarded after use.
+function cli.cleanup()
+  for k, v in pairs(package.loaded) do
 
-local split = function(str, pat)
-  local t = {}
-  local fpat = "(.-)" .. pat
-  local last_end = 1
-  local s, e, cap = str:find(fpat, 1)
-  while s do
-    if s ~= 1 or cap ~= "" then
-      table.insert(t,cap)
+    if (v == cli) or (k:match('cliargs')) then
+      package.loaded[k] = nil
+      break
```

Comment:
yes, i'm with @Tieske on this that having things in a single file made things much simpler, but it's harder to test and make things private that way...